src/transformers/models/whisper/tokenization_whisper.py

```diff
@@ -606,6 +606,10 @@ def _decode(
     ) -> str:
         self._decode_use_source_tokenizer = kwargs.pop("use_source_tokenizer", False)
 
+        has_prompt = isinstance(token_ids, list) and self.convert_tokens_to_ids("<|startofprev|>") in token_ids
```

Comment:
went back to using 'in' check here instead of indexing to 0 in token_ids since it errors if empty

Reply:
The reason I suggested putting it inside a separate `if skip_special_tokens:` is that the `in` operation needs to scan through the list, which is slow, and we can avoid it if `skip_special_tokens` is False.

Reply:
good point i'll adapt it to check the index 0